pkg/cmd/cluster/cluster_list.go

```diff
@@ -28,13 +30,42 @@ func newCmdClusterList(ctx api.Context) *cobra.Command {
 				currentClusterID = currentCluster.ID()
 			}
 
-			var clusters []*config.Cluster
-			if !onlyAttached {
-				clusters = ctx.Clusters()
-			} else if currentCluster != nil {
-				clusters = append(clusters, currentCluster)
+			if onlyAttached && currentCluster == nil {
+				return errors.New("no cluster is attached. Please run `dcos cluster attach <cluster-name>`")
```

Comment:
When there are no clusters configured, an empty file in `~/.dcos/dcos.toml` and I run `dcos cluster list --attached`, instead of seeing this message I have :
```
     NAME   ID      STATUS     VERSION  URL  
  *        .dcos  UNAVAILABLE  UNKNOWN
```

Reply:
https://jira.mesosphere.com/browse/DCOS-38927